devito/ir/iet/nodes.py

```diff
@@ -1122,14 +1127,13 @@ def _process_kwargs(cls, **kwargs):
 
         return kwargs
 
-
-class ParallelBlock(Block):
-
-    """
-    A sequence of Nodes, wrapped in a parallel block {...}.
-    """
-
-    is_ParallelBlock = True
+    @cached_property
+    def collapsed(self):
+        ret = [self]
+        for i in range(self.ncollapsed - 1):
+            ret.append(ret[i].nodes[0])
+        assert all(i.is_Iteration for i in ret)
```

Comment:
why not just `ret.append(ret[i].nodes[0] if ret[i].nodes[0]. is_Iteration)` ?

Reply:
because all of the nodes are expected to be Iterations at that point, and an assertion should cause a failure if that's not the case